src/event_listener/shared.rs

```diff
@@ -666,109 +669,307 @@ enum ParsedEventType {
+            r"openwindow>>(?P<address>.*),(?P<workspace>.*),(?P<class>.*),(?P<title>.*)",
+        ),
+        (
+            ParsedEventType::WindowClosed,
+            r"closewindow>>(?P<address>.*)",
+        ),
+        (
+            ParsedEventType::WindowMoved,
+            r"movewindow>>(?P<address>.*),(?P<workspace>.*)",
+        ),
+        (
+            ParsedEventType::LayoutChanged,
+            r"activelayout>>(?P<keyboard>.*)(?P<layout>.*)",
+        ),
+        (ParsedEventType::SubMapChanged, r"submap>>(?P<submap>.*)"),
+        (
+            ParsedEventType::LayerOpened,
+            r"openlayer>>(?P<namespace>.*)",
+        ),
+        (
+            ParsedEventType::LayerClosed,
+            r"closelayer>>(?P<namespace>.*)",
+        ),
+        (
+            ParsedEventType::FloatStateChanged,
+            r"changefloatingmode>>(?P<address>.*),(?P<floatstate>[0-1])",
+        ),
+        (
+            ParsedEventType::Minimize,
+            r"minimize>>(?P<address>.*),(?P<state>[0-1])",
+        ),
+        (
+            ParsedEventType::Screencast,
+            r"screencast>>(?P<state>[0-1]),(?P<owner>[0-1])",
+        ),
+        (
+            ParsedEventType::UrgentStateChanged,
+            r"urgent>>(?P<address>.*)",
+        ),
+        (
+            ParsedEventType::WindowTitleChanged,
+            r"windowtitle>>(?P<address>.*)",
+        ),
+        (ParsedEventType::Unknown, r"(?P<Event>^[^>]*)"),
+    ]
+    .map(|(e, r)| (e, check_for_regex_error(Regex::new(r))))
+});
+
+/// TODO: Possibly switch to this
+pub(crate) fn event_parser_v2(event: String) -> crate::Result<Vec<Event>> {
+    // TODO: Optimize nested looped regex capturing. Maybe pull in rayon if possible.
+    let event_iter = event
+        .trim()
+        .lines()
+        .map(|event_line| {
+            let type_matches = EVENT_SET
+                .iter()
+                .filter_map(|(event_type, regex)| Some((event_type, regex.captures(event_line)?)))
+                .collect::<Vec<_>>();
+
+            (event_line, type_matches)
+        })
+        .filter(|(_, b)| !b.is_empty());
+
+    let mut temp_event_holder = Vec::new();
+
+    for (event_str, matches) in event_iter {
+        match matches.len() {
+            0 => unreachable!(),
+            1 => {
+                report_unknown!((event_str.split('>').next().unwrap_or("unknown")));
+                continue;
+            }
+            2 => {
+                let (event_type, captures) = matches
+                    .into_iter()
+                    .find(|(e, _)| **e != ParsedEventType::Unknown)
+                    .unwrap_or_else(|| unreachable!());
+
+                temp_event_holder.push((event_str, event_type, captures));
+            }
+            _ => {
+                return Err(HyprError::IoError(io::Error::new(
+                    io::ErrorKind::InvalidData,
+                    "Event matched more than one regex (not a unknown event issue!)",
```

Comment:
Can u fix this typo, it should be `an` not `a`

Reply:
👍